File: test/test_firestore.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:firebase_core/firebase_core.dart';
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:flutter/widgets.dart';

void main() {
  testWidgets('Fetch users from Firestore', (WidgetTester tester) async {
    WidgetsFlutterBinding.ensureInitialized();
    await Firebase.initializeApp(
      options: const FirebaseOptions(
        apiKey: 'AIzaSyBhXMt6b8Ih44h6fQ1sCErS6mqExp8wJAI',
        appId: '1:762306964857:web:9cbdd52195759875c113a8',
        messagingSenderId: '762306964857',
        projectId: 'runa-f1e8e',
        authDomain: 'runa-f1e8e.firebaseapp.com',
        storageBucket: 'runa-f1e8e.firebasestorage.app',
        measurementId: 'G-VQX24PJMPJ',
      ),
    );

    final firestore = FirebaseFirestore.instance;
    final snapshot = await firestore.collection('users').get();
    print('TOTAL USERS: ${snapshot.docs.length}');
    for (var doc in snapshot.docs) {
      print('USER: ${doc.id} => ${doc.data()}');
    }
  });
}

File: test_firestore.mjs
import { initializeApp } from "firebase/app";
import { getFirestore, collection, getDocs } from "firebase/firestore";

const firebaseConfig = {
  apiKey: "AIzaSyB7Nz5o4XcZ1waXsKSEDiiSJRU_xt2oIzE",
  authDomain: "runaapp-cca6a.firebaseapp.com",
  projectId: "runaapp-cca6a",
  storageBucket: "runaapp-cca6a.firebasestorage.app",
  messagingSenderId: "787850092507",
  appId: "1:787850092507:web:04dbee19af71b0808b51fe",
  measurementId: "G-QWCSG6F5YR"
};

const app = initializeApp(firebaseConfig);
const db = getFirestore(app);

async function testRead() {
  console.log("Mencoba membaca koleksi 'users' menggunakan Client Key...");
  try {
    const querySnapshot = await getDocs(collection(db, "users"));
    console.log(`Berhasil! Ditemukan ${querySnapshot.size} users.`);
    querySnapshot.forEach((doc) => {
      console.log(doc.id, " => ", doc.data().username || doc.data().email);
    });
  } catch (error) {
    console.error("Gagal membaca Firestore:", error.message);
  }
}

testRead();
